test/fixtures/attribute-generation-classname-rewriting.js

```diff
@@ -36,6 +36,10 @@ export default () => {
       <div {...props} {...rest} />
       <div {...props} data-foo {...rest} />
       <div {...props} className={'test'} data-foo {...rest} />
+      <div {...{ className: 'test' }} />
+      <div {...{ className: 'test' }} data-test="test" />
+      <div {...{ className: 'test' }} {...props} data-test="test" />
+      <div {...props} {...{ id: 'test' }} />
```

Comment:
thanks can you add the other tests form the comment? It is important to make sure that the element gets a `className` prop the spread is merged correctly

Reply:
My bad, I did not see your tests cases 🙈

Reply:
I have updated test cases. 
Can you verify that it handles multiple classNames correctly?

Reply:
looking good! We should add one last test case to make sure that it is handled correctly:

```jsx
<div className="test" {...{ className: props.className }} />
<div className="test" {...{ className: props.className }} {...bar } />
<div className="test" {...bar }  {...{ className: props.className }}  />
```